### PR TITLE
Convert to cljc

### LINKED ISSUE #19: Switch from .cljx to .cljc
Still waiting on wider adoption of Clojure 1.7+

### PR DESCRIPTION
**Pull request by danielcompton**
Happy to hear your thoughts on this. This compiles in Clojure and ClojureScript without warnings, after 303b76e.

Fixes #19

### CODE REVIEW

File: src/taoensso/encore.cljc [hunk 1]
```diff
@@ -71,16 +71,17 @@
...
-  "Evaluates `test`. If it doesn't error and returns logical true, expands to
-  `then`, otherwise expands to `else`. Stolen from `clojure.core.reducers`.
-  (compile-if (Class/forName \"java.util.concurrent.ForkJoinTask\")
-    (do-cool-stuff-with-fork-join)
-    (fall-back-to-executor-services))"
-  [test then & [else]]
-  (if (try (eval test) (catch Throwable _ false))
-    `(do ~then)
-    `(do ~else)))
+#?(:clj
```
Comment (danielcompton) [inline src/taoensso/encore.cljc]: I think these reader conditionals are required now to make sure they're only executed when reading in the Clojure context. Before with cljx, there were two separate files created for ClojureScript, and the macros were only evaluated in the Clojure compilation context.

**Comment by ptaoussanis** [comment]
Thanks Daniel! Want to wait for the Clojure v1.9 alphas to stabilize a little, then will take a look at cutting a breaking release that includes your work here.

Cheers :-)

**Comment by ptaoussanis** [comment]
Hi Daniel,

Have had a few requests to retain Encore back compatibility if possible (Encore's often used as a shim for backported Clojure features). As I understand it, there's no particular motivation for a .cljc port here - right? (Compared to Sente, where you've got specific PRs that you'd like to submit - yes?).

For my part, am quite happy to keep putting this off until there's a compelling reason to break.

To be clear: am still on board for a .cljc port for Sente v1.10.

**Comment by ptaoussanis** [comment]
Closing for now (just doing issue triage); please feel free to reopen if I've misunderstood and you've got a specific motivation for getting this in with urgency.

Cheers :-)

**Comment by danielcompton** [comment]
No worries, I was keen to do a port as Cursive likes CLJC files more than CLJX, but it's no big deal.

**Pull request closed without merging**